deepmd/nvnmd/utils/network.py

```diff
@@ -67,39 +70,13 @@ def qr(x, nbit):
     y = x + tf.stop_gradient(y - x)
     return y
 
-
-# fitting_net
-def tanh2(x, nbit=-1, nbit2=-1):
-    r"""User-defined activation function tanh2
-
-    Parameter
-    ---------
-    x
-        input tensor
-    nbit
-        quantification precision for forward calculation
-    nbit2
-        quantification precision for backward calculation
-    """
-    y = op_module.tanh2_nvnmd(x, 0, nbit, nbit2, -1)
-    return y
-
-
-def tanh4(x, nbit=-1, nbit2=-1):
-    r"""User-defined activation function tanh4
-
-    Parameter
-    ---------
-    x
-        input tensor
-    nbit
-        quantification precision for forward calculation
-    nbit2
-        quantification precision for backward calculation
-    """
-    y = op_module.tanh4_nvnmd(x, 0, nbit, nbit2, -1)
-    return y
-
+def tanh4(x):
+    sign = tf.sign(x)
+    xclp = tf.clip_by_value(x, -2, 2)
+    xabs = tf.abs(xclp)
+    y1 = (1.0/16.0) * tf.pow(xabs, 4) + (-1.0/4.0) * tf.pow(xabs, 3) + xabs
+    y2 = y1 * sign
+    return y2
```

Comment:
For a function with multiple operators, I suggest adding a name scope:
```py
with tf.name_scope("some_name"):
    ...
```

Reply:
Thank you very much. We have made changes.